python/whylogs/experimental/core/udf_schema.py

```diff
@@ -198,6 +229,31 @@ def decorator_register(func):
     return decorator_register
 
 
+def register_type_udf(
+    col_type: DataType,
+    udf_name: Optional[str] = None,
+    metrics: Optional[List[MetricSpec]] = None,
+    namespace: Optional[str] = None,
+    schema_name: str = "",
+    anti_metrics: Optional[List[Metric]] = None,
+) -> Callable[[Any], Any]:
+    def decorator_register(func):
+        global _multicolumn_udfs, _resolver_specs
+        name = udf_name or func.__name__
+        name = f"{namespace}.{name}" if namespace else name
+        _multicolumn_udfs[schema_name].append(UdfSpec(None, {name: func}, col_type))
+        if metrics:
+            _resolver_specs[schema_name].append(ResolverSpec(None, col_type, deepcopy(metrics)))
```

Comment:
I think this will leak [anti-]metrics out onto other columns of the matched type. Since we don't know the input or output column names at registration time, we may not be able to support this...

Reply:
Is the same true for the metrics then?

do we need the anti_metrics on this decorator? Can't they also do that type->metric binding in the schema? I suppose that means generating the schema is trickier but it might be an acceptable escape hatch for this more advanced scenario.

Reply:
> Is the same true for the metrics then?

Assuming you mean metric UDFs, no. The [anti-]metrics registered there apply only to the submetric created by the UDF. Regular dataset UDFs are also OK, since the output column name is known at registration time. 

> do we need the anti_metrics on this decorator? 

I'm going to remove them. For this sort of UDF, you'll have to explicitly manage your resolver specs if you want something other than the unadulterated schema.